File: angular/base-bff-api/projects/bff-api/src/lib/models/langdesc.ts
// From BaseUtil - LangDesc
export class LangDesc {
  /** Bangladesh */
  bd?: string;
  /** English */
  en?: string;
  /** Indonesian */
  in?: string;
  /** Malaysia */
  my?: string;

  constructor(args?: {
      en?: string,
      bd?: string,
      in?: string,
      my?: string
    }) {

    if (args) {
      if (args.en) {
        this.en = args.en;
      }
      if (args.bd) {
        this.bd = args.bd;
      }
      if (args.in) {
        this.in = args.in;
      }
      if (args.my) {
        this.my = args.my;
      }
    }
  }

  setEn(desc: string) {
    this.en = desc;
  }

  setBd(desc: string) {
    this.bd = desc;
  }

  setIn(desc: string) {
    this.in = desc;
  }

  setMy(desc: string) {
    this.my = desc;
  }

  getLangDesc() {
    const langDesc = new LangDesc({en: ''});

    if (this.en) {
      langDesc.en = this.en;
    }
    if (this.bd) {
      langDesc.bd = this.bd;
    }
    if (this.in) {
      langDesc.in = this.in;
    }
    if (this.my) {
      langDesc.my = this.my;
    }
    return langDesc;
  }
}

File: base-util/src/main/java/com/util/model/LangDesc.java
package com.util.model;


import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;


/**
 * @author mary.jane
 *
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(Include.NON_NULL)
public class LangDesc implements Serializable {

	private static final long serialVersionUID = 8409225214310874187L;

	/** Bangladesh */
	private String bd;

	/** English */
	private String en;

	/** Indonesian */
	private String in;

	/** Malaysia */
	private String my;


	public LangDesc() {
		// DO NOTHING
	}


	public String getBd() {
		return bd;
	}


	public void setBd(String bd) {
		this.bd = bd;
	}


	public String getEn() {
		return en;
	}


	public void setEn(String en) {
		this.en = en;
	}


	public String getIn() {
		return in;
	}


	public void setIn(String in) {
		this.in = in;
	}


	public String getMy() {
		return my;
	}


	public void setMy(String my) {
		this.my = my;
	}

}
